PluginManager/PluginManager.cpp

```diff
@@ -277,26 +276,43 @@ namespace plugin
         return ResolvePathInternal(filename, "PLUGIN_PATH", "plugins/");
     }
 
-    std::filesystem::path PluginManager::ResolvePathInternal(std::string filename,
+    std::filesystem::path PluginManager::ResolvePathInternal(std::filesystem::path filename,
                                                const char* envVarOverride,
                                                std::string defaultPrefix)
     {
         if(filename.empty()) {
             throw std::logic_error{ "Received blank filename in ResolvePathInternal" };
         }
 
-        std::ifstream file(filename.c_str());
-
         // Respect absolute paths. We assume that client passed an absolute
         // path because they know the exact location which is fixed.
-        if (file.is_open() || filename[0] == '/') {
+        if(std::filesystem::exists(filename) || filename.is_absolute()) {
             return filename;
         }
 
         // Check environment variable to override the default location
         const char *pluginDir = std::getenv(envVarOverride);
+
+        // split pluginDir into a vector of strings on the : character, allows for multiple paths to be specified
```

Comment:
support multiple paths -- this is the primary purpose of this MR, to allow me to do something like:

```
PLUGIN_PATH=/home/rkerr/gitroot/applications-build-release/plugins:/home/rkerr/gitroot/algorithms-build-release/plugins
```

...and then I have backend and front-end plugins available directly from their build output folders rather than having to install both projects to get them all in /opt/pulmera/bin/plugins, which is where they'll be on the real system, which'll find them w/o the environemnt variable using the last code path starting on line 318.

Reply:
security risk ? should we have the prod app ignore env variables ?

Reply:
Added Notion task to the Cybersecurity "epic".